### ISSUE TITLE
Arilux Bulb?

### ISSUE THREAD

**Issue by packetinspector** [stelgenhof/AiLight#26]
I flashed the code on to one of these, but it just caused boot loops.

http://tinkerman.cat/yet-another-wifi-light-bulb/#more-2127

I'm going to try Espurna, but is there anything I might be able to change in your code to make it compatible?

**Comment by stelgenhof** [comment]
I am not aware these Arilux can work or not with the AiLight firmware. According to that link, it uses the same LED driver as the Ai-Thinker Light, so from that it looks like it should work.

Unfortunately I don't have an Arilux bulb, so cannot really test if it works or not. (https://github.com/stelgenhof/AiLight/wiki/Compatible-Light-Bulbs)

Espurna might work in this case, as this a more generic firmware designed for general ESP8266 based devices.

**Comment by packetinspector** [comment]
I suspect it has something to do with the flash/eeprom config.

**Comment by stelgenhof** [comment]
Yes, that's possible. Also not all the ESP8266 chips used in these bulbs are the same. Always hard to assess which bulb works and which doesn't unfortunately.

**Comment by stelgenhof** [comment]
@packetinspector Perhaps not related to your Arilux bulb, many have actually experienced reset loops (Software WDT reset) as well (including myself). I have made some fixes in the master branch which makes the firmware now work with Arduino ESP8266 v2.4.0 framework also.

Could you give it a try and let me know if it works or not?

Cheers! Sacha

**Comment by stelgenhof** [comment]
@packetinspector Just wanted to follow up if you had any success with the latest master branch code?

**Comment by xbmcnut** [comment]
I have just tried v5.13.1 on an Arilux bulb and while it flashed, the webpage does not load correctly preventing access to program any settings (WiFi, MQTT etc.)

 
![image](https://user-images.githubusercontent.com/1492137/39731324-2e49261a-52bb-11e8-813f-7ab617492caa.png)

**Comment by stelgenhof** [comment]
@xbmcnut Sorry I don't recognize this screenshot and neither the version number you are referring to (v5.13.1). What is this firmware you are trying to flash?

**Comment by xbmcnut** [comment]
Disregard. Seems I was connecting to another ESP unit running a MiLight bridge! Sorry about that. Don't know why that was up on 192.168.4.1 as it had been configured for WiFi already. My Arilux bulb does not seem to be working after flashing 0.5.0 release. No AP mode.

**Comment by stelgenhof** [comment]
Can't really tell what the issue is you are experiencing as I don't have an Arilux bulb myself. Not confirmed if the firmware works with this lightbulb or not. Best is to check the output of the Serial connector to see if there are any clues why it is not working.

**Comment by xbmcnut** [comment]
FYI, I tried espurna-1.12.6-arilux-e27.bin just now and the AP mode is exposed and I can connect and control [this bulb](https://www.banggood.com/ARILUX-E27-11W-RGBW-Dimmable-Smart-WIFI-LED-Light-Bulb-Works-with-Amazon-Alexa-Echo-AC85-265V-p-1175196.html?cur_warehouse=CN). FYI, the packaging (generic) says it's 7W not 11 as stated in the listing.

**Comment by stelgenhof** [comment]
Some lightbulbs actually use different LED driver chips to control the various LED's on the PCB. The AiLight firmware only works with the MY9291 LED driver, so any bulb using that driver will work.

I believe the AriLux is using a different driver. Other firmwares like Espurna or Tasmota are compatible with a variety of lightbulbs/drivers so that explains why they work :)

**Comment by xbmcnut** [comment]
I can confirm the Arilux bulb I have has a MY9291 LED driver. I'll reflash again this evening and see if it works 2nd time around.

**Comment by stelgenhof** [comment]
You're correct. The Arilux E27 is indeed using the MY9291 driver. Was mixing it with another bulb I had in mind. Let me know if you managed to flash it (or not).

**Comment by badgerhome** [comment]
Looking forward to seeing If you get this to work. Looks like they are 11w bulbs so would be interesting to see how brightness compares to the others.

**Comment by xbmcnut** [comment]
Tried re-flashing this evening and again no AP mode. Seems device is constantly rebooting. PuTTy on COM8 @ 115200 below.

```
 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 1384, room 16
tail 8
chksum 0x2d
csum 0x2d
v02f0c112
~ld

 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 1384, room 16
tail 8
chksum 0x2d
csum 0x2d
v02f0c112
~ld
```

Works fine though with espurna-1.12.6-arilux-e27.bin.

**Comment by stelgenhof** [comment]
Ah yes. It's same as the OP mentioned. Are you just flashing the binary release file, or compiling yourself?

**Comment by xbmcnut** [comment]
Binary release. Very rusty on Platformio.

**Comment by xbmcnut** [comment]
Binary on both instances. Ailight and Espurna.

**Comment by xbmcnut** [comment]
Should it work if I build with Platformio?

**Comment by xbmcnut** [comment]
SUCCESS! Compiled with Platformio and it is working a treat now. What's the difference between that and your compiled binaries?

**Comment by stelgenhof** [comment]
Good to hear! 

v0.5.0 (binary release) has this issue with repeated reboots experienced by a few users. If you compile from source code yourself you are always using the latest changes, in which I have resolved the issues from v0.5.0.

I will release a new version (+ binary) once the latest changes are stable.

**Comment by xbmcnut** [comment]
Ah yes, see this now. Build is running the version below which is working a treat. Thanks for all your hard work and advice.

![image](https://user-images.githubusercontent.com/1492137/39754646-f97b1a54-5316-11e8-82e3-8930e293f903.png)

**Comment by stelgenhof** [comment]
Super! Thanks for the testing and patience. My lightbulbs are working fine as well on this build. Guess I will prepare a release soon :)

**Comment by xbmcnut** [comment]
For those wishing to flash these four channel bulbs via MQTT, here is the json.

`{"flash":"10", "brightness":255,"white_value":0,"color_temp":500, "color": {"r": 255, "g": 0, "b": 0}}`

**Comment by xbmcnut** [comment]
I'm going to use the above when any of my smoke alarms goes off.

**Comment by stevesoltys** [comment]
Just here to confirm that I was able to flash the latest development build of AiLight on this bulb: https://www.amazon.com/gp/product/B077VL5JXR/

It appears to be the same as the Arilux E27 ([packaging and all](http://tinkerman.cat/wp-content/uploads/2017/11/20171109_142859s.jpg)) but without the branding on the side.

Thanks for the fix!

**Comment by stelgenhof** [comment]
Included in the just released version 0.6.0.
